### DISCUSSION TITLE
How to generate pbf from osm2pgsql generated postgres data.

### DISCUSSION THREAD

**Discussion by sawandarekar** [osm2pgsql-dev/osm2pgsql#1481] [Ideas]
By using osm2pgsql tool, i have imported the **PBF** data in **postgres**. 

Now i want to do reverse, Have to create pbf file from, what ever data is there in postgres.

I see the option as osmosis but, osm2pgsql and osmosis have different schema structure.

so osmosis did not work on osm2pgsql schema. Is there any other way for this.

**Comment by joto** [comment]
There is no way to do this and it doesn't make sense either. Osm2pgsql converts the data in complex ways into something that is useful, for instance, for rendering. In that process a lot of information is lost. If you put a banana into a blender you might get a nice banana smoothie, but you'll not get your banana back from the smoothie. For that kind of thing you need a different database format, for instance the one osmosis provides.

**Comment by sawandarekar** [comment]
Thanks for details...will check another ways.
